rdrf/rdrf/admin.py

```diff
@@ -255,11 +250,19 @@ def activate_users(self, request, queryset):
         for profile in queryset:
             self.activate_user(profile.activation_key)
 
+    def get_registration_activation_url(self, registration_profile):
```

Comment:
Why is this required?

Reply:
Because the `send_activation_email` function in PatientRegistration has code `self.get_registration_activation_url(registration)`. When we call `send_activation_email` function here, the self points to the CustomRegistrationProfileAdmin class, which then require CustomRegistrationProfileAdmin class has a `get_registration_activation_url` function, otherwise, an error will pop up. Same reason for adding `custom_activation_key_expired` function in PatientActivationView.

Reply:
`PatientRegistration` is a class that should be instantiated, and the `send_activation_email` called on the instantiated object. e.g.
```
            patient_registration = PatientRegistration(request)
            patient_registration.send_activation_email(user.registry_code, user, user.patient)
```

Even so, we shouldn't be naming the registration class directly, as it can be customised per registry. It is generally "loaded" via the `registration_rdrf`'s `load_registration_class`.

At this point I'm wondering how this is going to work for MND where patients have been added by clinicians (i.e. NOT self registered), and also what it will look like for carers.

@ScriptSmith Do you think it's worth taking this solution further to address the above, or do we just call it at this point to say it's too complicated, and remove the option to "resend" the activation email?